### PR TITLE
https support

### PR DESCRIPTION
**Pull request by nofateg**

### CODE REVIEW

File: oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java [hunk 1]
```diff
@@ -151,31 +184,68 @@ private void startNewPort( int port ) {
...
+                Metrics.gauge( "nio_pool_size", Tags.of( "port", sPort, "schema", schema.name(), "name", "worker", "type", "core" ), server, s -> s.getWorker().getMXBean().getCoreWorkerPoolSize() );
+                Metrics.gauge( "nio_pool_size", Tags.of( "port", sPort, "schema", schema.name(), "name", "worker", "type", "max" ), server, s -> s.getWorker().getMXBean().getMaxWorkerPoolSize() );
+                Metrics.gauge( "nio_pool_size", Tags.of( "port", sPort, "schema", schema.name(), "name", "worker", "type", "busy" ), server, s -> s.getWorker().getMXBean().getBusyWorkerThreadCount() );
+                Metrics.gauge( "nio_pool_size", Tags.of( "port", sPort, "schema", schema.name(), "name", "worker", "type", "queue" ), server, s -> s.getWorker().getMXBean().getWorkerQueueSize() );
 
             }
         }
     }
 
+    private synchronized SSLContext getSSLContext() {
```
Comment (stuart12) [inline oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java:203]: I do not think that you need to create a SSLContext as you can just pass the manager to `builder.addHttpsListener( port, "0.0.0.0", managers, null, handler )`

File: oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java [hunk 2]
```diff
@@ -211,20 +210,18 @@ private synchronized SSLContext getSSLContext() {
...
             kmf.init( ks, sslConfiguration.password.toCharArray() );
 
-            sslContext.init( kmf.getKeyManagers(), null, null );
+            keyManagers = kmf.getKeyManagers();
         } catch( IOException e ) {
             throw new UncheckedIOException( e );
-        } catch( UnrecoverableKeyException | CertificateException | KeyStoreException | NoSuchAlgorithmException |
-                 KeyManagementException e ) {
+        } catch( UnrecoverableKeyException | CertificateException | KeyStoreException | NoSuchAlgorithmException e ) {
             throw new RuntimeException( e );
```
Comment (stuart12) [inline oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java:221]: do we really want to throw a RuntimeException? If the http server is configured to do both HTTP and HTTPS, it should do HTTP even if the HTTPS setup fails. NioHttpServer though only throw a RuntimeException if HTTPS setup fails and it is not configured to do HTTP.

Comment (nofateg) [reply oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java:221]: It's strange when we configure http and https and the server starts with a missing certificate for example.

Comment (stuart12) [reply oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java:221]: I agree that the best answer is not obvious. I feel however that if you ask the code to do HTTP and HTTPS but it cannot do HTTPS, then if should log an error and do HTTP while I fix the problem with the keystore. My use case is that I'm adding HTTPS to a system currently working with just HTTP,  and I'd like to be sure not to break HTTP while I'm adding HTTPS.

Comment (nofateg) [reply oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java:221]: We test the **https** implementation with tests and it either works or the tests fail. The only thing we don't test with unit tests is the working configuration, but that's not the case with this code.

Comment (stuart12) [reply oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java:221]: as you wish, however your tests will not stop the user from providing a bad (or missing) keystore.jks. In this case throwing a runtimexception will break a valid HTTP configuration if HTTPS is also requested

**Pull request closed without merging**